### PR TITLE
Run integration tests vs production

### LINKED ISSUE #270: Add continuous integration build that runs the integration tests against production.
This probably needs to be inside Google so we can use the right credentials.

### PR DESCRIPTION
**Pull request by coryan**
This fixes #270.  Runs the integration builds against production for our Jenkins builds.  Caught a couple of test bugs along the way.

### CODE REVIEW

File: bigtable/tests/integration_tests_utils.sh [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+  local project_id=$1
+  shift
+  local instance_id=$1
+  shift
+
+  # If the instance is not set, just return immediately.
+  if [ -z "${instance_id}" ]; then
+    return 0
+  fi
+  ${CBT_CMD} -project "${project_id}" -instance "${instance_id}" ls |
```
Comment (sduskis) [inline bigtable/tests/integration_tests_utils.sh]: The instance may be used by other tests that may need long lived tables.  For now, can you please add a TODO for a more targeted strategy in the future?

File: ci/build-docker.sh [hunk 2]
```diff
@@ -78,7 +78,7 @@ CTEST_OUTPUT_ON_FAILURE=1 make -j ${NCPU} test
 if [ -r /etc/lsb-release ] && grep -q 14.04 /etc/lsb-release; then
   echo "Skipping integration tests, Go version too old in Ubuntu 14.04."
 else
-  (cd bigtable/tests && ../../../../bigtable/tests/run_integration_tests.sh)
+  (cd bigtable/tests && /v/bigtable/tests/run_integration_tests_emulator.sh)
```
Comment (sduskis) [inline ci/build-docker.sh]: How does /v/ work here?

Comment (sduskis) [reply ci/build-docker.sh]: Oh, I see. It's a docker thing.

File: bigtable/tests/integration_tests_utils.sh (hunk 1 above)
Comment (coryan) [reply bigtable/tests/integration_tests_utils.sh]: Note that I created a separate instance to run the integration tests so the other tests are not affected.  I also created a lock in Jenkins to represent this new instance.

In any case, I added #356 to make the integration tests smarter, and put the TODO linking to the new issue in the script.

File: ci/build-docker.sh (hunk 2 above)
Comment (coryan) [reply ci/build-docker.sh]: Yup.

**Pull request merged**